Server/scripts/fetchYearMaps.ts

```diff
@@ -51,33 +51,26 @@ const getModeDivison = memoizee(async (modeDivisionId: number) => {
     return mode;
 });
 
-// Memoized method to create or fetch a BeatmapSet from DB.
-const existingSets: number[] = [];
-const getBeatmapSet = memoizee(async (beatmap: APIBeatmap): Promise<Beatmapset> => {
-    if(existingSets.includes(beatmap.setId))
-        return await Beatmapset.findOne(beatmap.setId) as Beatmapset;
-    let beatmapSet = new Beatmapset;
-    beatmapSet.ID = beatmap.setId;
-    beatmapSet.approvedDate = beatmap.approvedDate;
-    beatmapSet.submitDate = beatmap.submitDate;
-    beatmapSet.BPM = beatmap.bpm;
-    beatmapSet.artist = beatmap.artist;
-    beatmapSet.title = beatmap.title;
-    beatmapSet.genre = genres[beatmap.genre];
-    beatmapSet.language = langs[beatmap.language];
-    beatmapSet.tags = beatmap.tags.join(" ");
-    beatmapSet.favourites = beatmap.favoriteCount;
+// API call to fetch a user's country.
+async function getUserCountry (userID: number): Promise<string> {
+    const userApi = (await axios.get(`https://osu.ppy.sh/api/get_user?k=${config.osu.v1.apiKey}&u=${userID}&type=id`)).data as any[];
+    if (userApi.length === 0)
+        return "";
+    return userApi[0].country;
+};
 
-    let user = await User.findOne({ osu: { userID: `${beatmap.creatorId}` } });
+const getUser = memoizee(async (targetUser: { username: string, userID: number, country?: string }): Promise<User> =>{
```

Comment:
Is `country` used here?

Reply:
line 71 https://github.com/Corsace/Corsace/blob/bn-info/Server/scripts/fetchYearMaps.ts#L71

```ts
user.country = targetUser.country ?? await getUserCountry(targetUser.userID);
```

the memoizee uses the username not the whole data structure (i think) based on line 95 https://github.com/Corsace/Corsace/blob/bn-info/Server/scripts/fetchYearMaps.ts#L95

Reply:
Ah right, I totally missed that is was unpacking the `targetUser` object, and I missed the `normalizer` as well. Thanks for pointing it out.